app/frontend/components/shared/permit-applications/requirement-form.tsx

```diff
@@ -163,7 +185,19 @@ export const RequirementForm = observer(
       if (onCompletedBlocksChange) {
         onCompletedBlocksChange(getCompletedBlocksFromForm(containerComponent.root))
       }
-      setErrorBoxData(mapErrorBoxData(containerComponent.root.errors))
+      // setErrorBoxData(mapErrorBoxData(containerComponent.root.errors))
+    }
+    const onScroll = (event) => {
+      if (hasErrors) {
+        if (isFirstComponentNearTopOfView(firstComponentKey)) {
+          setFloatErrorBox(true)
+        }
+        if (!isFirstComponentNearTopOfView(firstComponentKey)) {
+          setFloatErrorBox(false)
+        }
+      } else {
+        setFloatErrorBox(false)
```

Comment:
This can be simplified to ` setFloatErrorBox(hasErrors && isFirstComponentNearTopOfView(firstComponentKey))`